### PR TITLE
[v2] update endpoints to separate intake from assets

### PR DESCRIPTION
**Pull request by roncohen**
Closes https://github.com/elastic/apm-server/issues/1336.

### CODE REVIEW

File: beater/route_config.go [hunk 1]
```diff
@@ -36,20 +36,28 @@ import (
 )
 
 var (
-	rootURL                   = "/"
+	rootURL        = "/"
+	HealthCheckURL = "/healthcheck"
+
+	// intake v1
```
Comment (simitt) [inline beater/route_config.go]: Those URLs are also deprecated.

File: beater/route_config.go [hunk 2]
```diff
@@ -36,20 +36,28 @@ import (
 )
 
 var (
-	rootURL                   = "/"
+	rootURL        = "/"
+	HealthCheckURL = "/healthcheck"
```
Comment (simitt) [inline beater/route_config.go]: will also be deprecated, as `rootUrl` is the new healthcheck URL.

**Comment by watson** [comment]
Consider adding deprecation warnings already now for the old sourcemap urls that will be removed in v7

**Comment by roncohen** [comment]
set sourcemap url to `/assets/v1/sourcemaps` and updated comments to reflect that all intake v1 urls are deprecated.

**Comment by roncohen** [comment]
@watson @simitt 👍 ?

File: docs/sourcemap-api.asciidoc [hunk 3]
```diff
@@ -35,7 +35,7 @@ Example source map request
 
 ["source","sh",subs="attributes"]
 ---------------------------------------------------------------------------
-curl -X POST http://127.0.0.1:8200/v1/rum/sourcemaps \
+curl -X POST http://127.0.0.1:8200/assets/v1/sourcemaps \
```
Comment (simitt) [inline docs/sourcemap-api.asciidoc]: I'd expect to have a note about the deprecated URL somewhere in the docs. It is going to be part of a minor release, we should avoid simply removing deprecated but supported URLS.

**Comment by roncohen** [comment]
> Didn't you have a deprecation warning somewhere in the old commits? Why did you remove that?

I don't think there was a deprecation warning, but I'm happy to add one.

**Comment by roncohen** [comment]
screen shot 2018-09-12 at 3 30 43 pm

**Comment by watson** [comment]
FYI, there's [a custom keyword](https://github.com/elastic/docs#additions-and-deprecations) you can use in asciidoc to show deprecation notices:

```asciidoc
deprecated[6.5.0,The URL for uploading sourcemaps was previously `/v1/sourcemaps`. It still works, but is being deprecated in favor of `/assets/v1/sourcemaps`.]
```

I use it in the Node.js docs here: https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#apm-build-span

**Comment by watson** [comment]
But come to think of it, that might look weird in this case. I'm also fine with how it looks now :)

**Comment by roncohen** [comment]
@elastic/apm-agent-devs heads up: this changes the intake v2 urls

**Comment by roncohen** [comment]
thanks @watson @simitt !

**Comment by mikker** [comment]
New one is `"/intake/v2/events"`?

**Comment by simitt** [comment]
yes @mikker

**Pull request merged**